datafusion/sql/src/planner.rs

```diff
@@ -600,27 +602,54 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
             SQLDataType::Array(ArrayElemTypeDef::AngleBracket(inner_sql_type)) => {
                 // Arrays may be multi-dimensional.
                 let inner_data_type = self.convert_data_type(inner_sql_type)?;
-                Ok(DataType::new_list(inner_data_type, true))
+
+                // Lists are allowed to have an arbitrarily named field;
```

Comment:
Should we create an issue for creating `DataType::new_list_from_field` in arrow-rs?